handle.go

```diff
@@ -3,76 +3,52 @@ package main
 import (
 	"fmt"
 	"sync"
-	"time"
 
-	pb "github.com/gautamrege/gochat/api"
+	"github.com/gautamrege/gochat/api"
 )
 
-type User struct {
-	pb.Handle
-	Created_at time.Time
-}
-
 // Ensure that users are added / removed using a mutex!
-type UserSync struct {
+type PeerHandleMapSync struct {
 	sync.RWMutex
-	UserMap map[string]User
+	PeerHandleMap map[string]api.Handle
 }
 
-var ME pb.Handle
-var USERS UserSync
-
-// Insert user if not exists
-// if existst then update the data
-func (hs *UserSync) Insert(h pb.Handle) (err error) {
+// Insert user if not exists already then add it
+func (hs *PeerHandleMapSync) Insert(newHandle api.Handle) (err error) {
 	hs.Lock()
-	_, ok := hs.UserMap[h.Name]
-	hs.UserMap[h.Name] = User{
-		Handle: pb.Handle{
-			Name: h.Name,
-			Port: h.Port,
-			Host: h.Host,
-		},
-		Created_at: time.Now(),
-	}
+	_, ok := hs.PeerHandleMap[newHandle.Name]
 	if !ok {
-		fmt.Printf("\nNew User joined the chat: @%s\n> ", h.Name)
+		hs.PeerHandleMap[newHandle.Name] = newHandle
```

Comment:
This code be above and before the condition. Consider the case where someone restarts the chat with the same handle but a different port. So, this also serves as an 'update' on handles. 

```suggestion
		hs.PeerHandleMap[newHandle.Name] = newHandle
		if !ok {
```